### PR TITLE
containerd: properly populate /etc/hosts and /etc/hostname

### LINKED ISSUE #6811: containerd not populating /etc/hostname
## Summary

Running `hostname --fqdn` returns an error when using the containerd runtime. It does not return an error when using guardian.

## Steps to reproduce

Bring up concourse with a worker using `CONCOURSE_RUNTIME: containerd`.

Bring up a container
```
fly -t dev sp -p test -c topgun/pipelines/task-waiting.yml
fly -t dev up -p test
fly -t dev tj -j test/simple-job
```
Intercept the container and run `hostname --fqdn`.

## Expected results

A hostname is printed.

## Actual results

Error is thrown
```
hostname: Name or service not known
```

## Additional context

Error is returned because `/etc/hosts` is not populated. If you look at `/etc/hosts` on a container started with containerd you'll see
```
$ cat /etc/hosts
127.0.0.1 localhost
```
While with guardian:
```
$ cat /etc/hosts
127.0.0.1 localhost
10.80.0.10 fd3610f2-5c06-42a0-669b-a8a437d1a9ca
```

Related PR https://github.com/containerd/cri/pull/1042

## Triaging info

* Concourse version:
* Browser (if applicable):
* Did this used to work?

### PR DESCRIPTION
**Pull request by muntac**
[add yourself]: https://github.com/concourse/governance#individual-contributors
[PR requirements]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md#pull-request-requirements

## Changes proposed by this PR
cc: @chenbh 

closes #6811

* [x] add hostname to `/etc/hostname`
* [x] add container's IP and hostname to `/etc/hosts`

## Notes to reviewer

The container's IP is only available once the network has been initialized. Therefore, the original /etc/hosts has to be updated after. The /etc/hosts on the container is bindmounted to a file on the worker.  We simply update this
file directly.

To test simply jump on to a running container and open the `/etc/hosts` and `/etc/hostname` file. You can also run `hostname -f` and `hostname -i` to validate.

## Release Note

* containerd: `/etc/hosts` and `/etc/hostname` are correctly populated

[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

### CODE REVIEW

File: worker/runtime/integration/integration_test.go [hunk 1]
```diff
@@ -434,6 +434,65 @@ func (s *IntegrationSuite) TestContainerAllowsHostAccess() {
...
+			s.containerdSocket(),
+			namespace,
+			requestTimeout,
+		),
+		networkOpt,
+	)
+	s.NoError(err)
+
+	s.NoError(customBackend.Start())
+
```
Comment (taylorsilva) [inline worker/runtime/integration/integration_test.go:457]: We don't need to create a custom garden backend if we're not changing anything from the default values. Use `s.gardenBackend` instead.
```suggestion
```

File: worker/runtime/integration/integration_test.go [hunk 2]
```diff
@@ -434,6 +434,65 @@ func (s *IntegrationSuite) TestContainerAllowsHostAccess() {
...
+	handle := uuid()
+
+	container, err := customBackend.Create(garden.ContainerSpec{
```
Comment (taylorsilva) [inline worker/runtime/integration/integration_test.go:460]:
```suggestion
	container, err := s.gardenBackend.Create(garden.ContainerSpec{
```

File: worker/runtime/integration/integration_test.go [hunk 3]
```diff
@@ -434,6 +434,65 @@ func (s *IntegrationSuite) TestContainerAllowsHostAccess() {
...
+		Handle:     handle,
+		RootFSPath: "raw://" + s.rootfs,
+		Privileged: true,
+	})
+	s.NoError(err)
+
+	defer func() {
+		s.NoError(customBackend.Destroy(handle))
+		customBackend.Stop()
```
Comment (taylorsilva) [inline worker/runtime/integration/integration_test.go:469]:
```suggestion
		s.NoError(s.gardenBackend.Destroy(handle))
```

File: worker/runtime/integration/integration_test.go [hunk 4]
```diff
@@ -434,6 +434,65 @@ func (s *IntegrationSuite) TestContainerAllowsHostAccess() {
...
+			Stdout: buf,
+			Stderr: buf,
+		},
+	)
+	s.NoError(err)
+
+	exitCode, err := proc.Wait()
+	s.NoError(err)
+
+	fmt.Println(buf.String())
```
Comment (taylorsilva) [inline worker/runtime/integration/integration_test.go:490]: from debugging I'm guessing
```suggestion
```

**Review by taylorsilva — CHANGES_REQUESTED** [review]

**Review by taylorsilva — APPROVED** [review]
LGTM

**Comment by muntac** [comment]
Thanks! Good eye with the test cleanup suggestions.

**Comment by jaredstehler** [comment]
would it be possible to get this included in the next point release?

**Comment by steveburton4** [comment]
@taylorsilva Do you know when this is planned to be released?

**Comment by taylorsilva** [comment]
Soon maybe? Looking at our milestone: https://github.com/concourse/concourse/milestone/75
It looks like we could probably cut another release soon.

**Comment by Christian-Oleson** [comment]
@jaredstehler , 
If I am reading this right it looks like this fix went out in the last release:
https://github.com/concourse/concourse/releases/tag/v7.4.0

**Comment by steveburton4** [comment]
> @jaredstehler ,
> If I am reading this right it looks like this fix went out in the last release:
> https://github.com/concourse/concourse/releases/tag/v7.4.0

It did. We have installed it already and tested it fixes the issue we were having 👍

**Pull request merged**